### PR TITLE
doc patterns requirement in `unnecessary_breaks`

### LINKED ISSUE #4131: Mention in the docs for unnecessary_breaks that it's a Dart 3.0 feature
The [docs for unnecessary_breaks](https://dart.dev/tools/linter-rules#unnecessary_breaks) should mention that they are enforcing a Dart 3.0 feature. The lint has no effect on code with sdk constraints lower than 3.0.0.

### PR DESCRIPTION
**Pull request by pq**
Fixes #4131

Feedback, word-smithing welcome.

/cc @goderbauer @bwilkerson @parlough

### CODE REVIEW

**Review by bwilkerson — APPROVED** [review]

File: lib/src/rules/unnecessary_breaks.dart [hunk 1]
```diff
@@ -54,6 +54,10 @@ switch (1) {
     print("just two");
 }
 ```
+
+NOTE: this lint only reports unnecessary breaks in libraries opted in to the 
+Dart 3.0 patterns language feature. Explicit breaks are still required in Dart
+2.19 and below.
```
[the remark this answers is not in the document]
Comment (parlough) [reply lib/src/rules/unnecessary_breaks.dart:60]: Could more explicitly mention "language version" and link off of that. Something like?

```
NOTE: This lint only reports unnecessary breaks in libraries with a
[language version](https://dart.dev/guides/language/evolution#language-versioning) of 3.0 or greater.
Explicit breaks are still required in Dart 2.19 and below.
```

[the remark this answers is not in the document]
Comment (pq) [reply lib/src/rules/unnecessary_breaks.dart]: Ah, very nice.

**Comment by pq** [comment]
@bwilkerson: take another look?  (Happy to be explicit about the patterns feature too.)

**Review by bwilkerson — APPROVED** [review]
I don't think there's any reason to be explicit about the language feature's name. It isn't important to users after it's been enabled by default.

**Comment by pq** [comment]
Super. Thanks for the help!

**Pull request merged**